plugin/core/registry.py

```diff
@@ -122,6 +122,19 @@ def session(self) -> Optional[Session]:
         else:
             return None
 
+    def sessions(self) -> List[Session]:
+        wm = windows.lookup(self.window)
+        if not wm:
+            return []
+        sessions = []  # type: List[Session]
+        for session in wm.get_sessions():
+            if self.capability and not session.has_capability(self.capability):
+                continue
+            if self.session_name and session.config.name != self.session_name:
```

Comment:
I don't think the addition of this if statement was necessary.

If you you know the session_name, you would probably use the `session_by_name` instead of this `sessions` method.

Reply:
maybe I don't understand something but if you have a command that overrides class property `session_name` then shouldn't `sessions()` respect that instead of returning all sessions? I don't think this method should be an override that would return all sessions in this case.

Reply:
It should

Reply:
Sorry, I wasn't too descriptive with my last message. 
I meant, It should respect. You were right.

If a command specified the session_name, session should respect that and just return that session in the array.